SignalServiceKit/src/Messages/OWSReadReceiptManager.m

```diff
@@ -445,6 +412,148 @@ - (void)updateOutgoingMessageFromLinkedDevice:(TSOutgoingMessage *)message
     [TSRecipientReadReceipt removeRecipientIdsForTimestamp:message.timestamp transaction:transaction];
 }
 
+#pragma mark - Linked Device Read Receipts
+
+- (void)updateIncomingMessage:(TSIncomingMessage *)message
+                  transaction:(YapDatabaseReadWriteTransaction *)transaction
+{
+    OWSAssert(message);
+    OWSAssert(transaction);
+
+    NSString *senderId = message.messageAuthorId;
+    uint64_t timestamp = message.timestamp;
+    if (senderId.length < 1 || timestamp < 1) {
+        OWSFail(@"%@ Invalid incoming message: %@ %llu", self.tag, senderId, timestamp);
+        return;
+    }
+
+    OWSLinkedDeviceReadReceipt *_Nullable readReceipt = [OWSLinkedDeviceReadReceipt linkedDeviceReadReceiptWithSenderId:senderId
+                                                                                                              timestamp:timestamp];
+    if (!readReceipt) {
+        return;
+    }
+    [message markAsReadWithTransaction:transaction sendReadReceipt:NO updateExpiration:YES];
+    [readReceipt removeWithTransaction:transaction];
+    
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [[NSNotificationCenter defaultCenter]
```

Comment:
Maybe lost in the myriad of PR's, but isn't this dispatch now redundant with `postNotificationNameAsync`?

Reply:
Done.